src/modules/vtol_att_control/tiltrotor.cpp

```diff
@@ -218,47 +231,77 @@ void Tiltrotor::process_mc_data()
 		set_idle_fw();
 		flag_idle_mc = false;
 	}
- }
 
-void Tiltrotor::process_fw_data()
-{
-	fill_att_control_output();
-}
+	_mc_roll_weight = 0.0f;
+	_mc_pitch_weight = 0.0f;
+	_mc_yaw_weight = 0.0f;
+ }
 
 void Tiltrotor::update_transition_state()
 {
+	if (!_flag_was_in_trans_mode) {
+		// save desired heading for transition and last thrust value
+		_yaw_transition = _v_att->yaw;
+		_throttle_transition = _actuators_mc_in->control[actuator_controls_s::INDEX_THROTTLE];
+		_flag_was_in_trans_mode = true;
+	}
 	if (_vtol_schedule.flight_mode == TRANSITION_FRONT_P1) {
+		// for the first part of the transition the rear rotors are enabled
+		if (_rear_motors != ENABLED) {
+			set_rear_motor_state(ENABLED);
+		}
 		// tilt rotors forward up to certain angle
-		if (_params_tiltrotor.front_trans_dur <= 0.0f) {
-			_tilt_control = _params_tiltrotor.tilt_transition;
-		} else if (_tilt_control <= _params_tiltrotor.tilt_transition) {
-			_tilt_control = _params_tiltrotor.tilt_mc + fabsf(_params_tiltrotor.tilt_transition - _params_tiltrotor.tilt_mc) * 
-						(float) hrt_elapsed_time(&_vtol_schedule.transition_start) / (_params_tiltrotor.front_trans_dur * 1000000.0f);
+		if (_tilt_control <= _params_tiltrotor.tilt_transition) {
+			_tilt_control = _params_tiltrotor.tilt_mc +  fabsf(_params_tiltrotor.tilt_transition - _params_tiltrotor.tilt_mc)*(float)hrt_elapsed_time(&_vtol_schedule.transition_start)/(_params_tiltrotor.front_trans_dur*1000000.0f);
 		}
 
 		// do blending of mc and fw controls
-		if (_airspeed->true_airspeed_m_s >= ARSP_BLEND_START && _params_tiltrotor.airspeed_trans - ARSP_BLEND_START > 0.0f) {
```

Comment:
Why did you remove this check?Are you sure that airspeed_trans > ARSP_BLEND_START?